xwiki-commons-core/xwiki-commons-velocity/src/main/java/org/xwiki/velocity/ScriptVelocityContext.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.velocity;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.script.Bindings;
+import javax.script.ScriptContext;
+
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.runtime.directive.Scope;
+
+/**
+ * Maintains the current ScriptContext in sync with any modification of the VelocityContext.
+ * 
+ * @version $Id$
+ * @since 15.9-rc-1
+ */
+public class ScriptVelocityContext extends XWikiVelocityContext
```

Comment:
Missing unstable on that one.

Reply:
Alors it feels like you could have a unit test for that one.

Reply:
Right, looks like I forgot to move the unit test from platform along with ScriptVelocityContext.

Reply:
Could not find any direct unit test in platform for the feature covered by ScriptVelocityContext (there might be some in the various other script macros tests), so I added one.